Doesn't support multiple resources in the same YAML

**Describe the bug**
`kubectl apply` supports YAML files that contain more than one Kubernetes resource manifest, separated by `---` lines. Moreover, this is quite common practice in the Kubernetes community. When I try to use such a YAML file with this terraform provider, I get an error message from terraform that says something like:

```
Error: expected to create 1 resource, got 12
```

**Steps to reproduce the issue:**
Try to use a multi-resource YAML manifest [like this one](https://raw.githubusercontent.com/kubernetes/ingress-nginx/nginx-0.24.1/deploy/mandatory.yaml) as the content for this terraform provider.

**Expected behavior**
Expected to `kubectl apply` all of the resources in the YAML manifest, but got an error instead.

**Additional context**
On Linux, using release [v0.2.0](https://github.com/banzaicloud/terraform-provider-k8s/releases/tag/v0.2.0).

small warning: the plugin crashes when one section is empty (I didn't take the time to test it, but my file had 3 valid sections, then the 4th one was entirely comments):
```
k8s_manifest.api[3]: Creating...

Error: rpc error: code = Unavailable desc = transport is closing



Error: rpc error: code = Unavailable desc = transport is closing



Error: rpc error: code = Unavailable desc = transport is closing



Error: rpc error: code = Unavailable desc = transport is closing


panic: runtime error: invalid memory address or nil pointer dereference
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16d331a]
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: 
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: goroutine 305 [running]:
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetNamespace(...)
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2:      /home/circleci/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/apis/meta/v1/unstructured/unstructured.go:234
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: github.com/banzaicloud/terraform-provider-k8s/k8s.resourceK8sManifestCreate(0xc0001865b0, 0x177a540, 0xc000456550, 0x2, 0x2b64d40)
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2:      /home/circleci/project/k8s/resource_k8s_manifest.go:62 +0x29a
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000270100, 0xc00066c460, 0xc00070a860, 0x177a540, 0xc000456550, 0xc0000d5301, 0xc0005d2480, 0x40eac3)
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2:      /home/circleci/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.4.0/helper/schema/resource.go:305 +0x365
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000270180, 0xc000947918, 0xc00066c460, 0xc00070a860, 0xc0001e4868, 0xc00070c018, 0x1847c20)
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2:      /home/circleci/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.4.0/helper/schema/provider.go:294 +0x99
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00026e050, 0x1e1fd40, 0xc000249b30, 0xc0000aa660, 0xc00026e050, 0xc000249b30, 0xc000906a80)
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2:      /home/circleci/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.4.0/internal/helper/plugin/grpc_provider.go:885 +0x882
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1a20a80, 0xc00026e050, 0x1e1fd40, 0xc000249b30, 0xc0000aa600, 0x0, 0x1e1fd40, 0xc000249b30, 0xc0006b4600, 0x595)
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2:      /home/circleci/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.4.0/internal/tfplugin5/tfplugin5.pb.go:3189 +0x217
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000e8000, 0x1e3df60, 0xc0000b7080, 0xc0006f2200, 0xc000248c30, 0x2b2e1e0, 0x0, 0x0, 0x0)
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2:      /home/circleci/go/pkg/mod/google.golang.org/grpc@v1.23.1/server.go:995 +0x460
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: google.golang.org/grpc.(*Server).handleStream(0xc0000e8000, 0x1e3df60, 0xc0000b7080, 0xc0006f2200, 0x0)
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2:      /home/circleci/go/pkg/mod/google.golang.org/grpc@v1.23.1/server.go:1275 +0xd97
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0003341b0, 0xc0000e8000, 0x1e3df60, 0xc0000b7080, 0xc0006f2200)
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2:      /home/circleci/go/pkg/mod/google.golang.org/grpc@v1.23.1/server.go:710 +0xbb
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-02-26T18:43:46.770+0100 [DEBUG] plugin.terraform-provider-k8s_v0.7.2:      /home/circleci/go/pkg/mod/google.golang.org/grpc@v1.23.1/server.go:708 +0xa1
2020-02-26T18:43:46.773+0100 [DEBUG] plugin: plugin process exited: path=/home/thomas/Downloads/terraform-provider-k8s_v0.7.2 pid=10245 error="exit status 2"
```

--- a/.gitignore
+++ b/.gitignore
@@ -22,3 +22,4 @@
 terraform.tfstate
 terraform.tfstate.backup
 .terraform/
+crash.log

--- a/README.md
+++ b/README.md
@@ -50,7 +50,7 @@ provider "k8s" {
 }
 ```
 
-The k8s Terraform provider introduces a single Terraform resource, a `k8s_manifest`. The resource contains a `content` field, which contains a raw manifest.
+The `k8s` Terraform provider introduces a single Terraform resource, a `k8s_manifest`. The resource contains a `content` field, which contains a raw manifest in JSON or YAML format.
 
 ```hcl
 variable "replicas" {
@@ -125,6 +125,24 @@ $ kubectl get deployments
 No resources found.
 ```
 
+**NOTE**: If the YAML formatted `content` contains multiple documents (separated by `---`) only the first non-empty document is going to be parsed. This is because Terraform is mostly designed to represent a single resource on the provider side with a Terraform resource:
+
+> resource types correspond to an infrastructure object type that is managed via a remote network API
+> -- <cite>[Terraform Documentation](https://www.terraform.io/docs/configuration/resources.html)</cite>
+
+You can workaround this easily with the following snippet (however we still suggest to use separate resources):
+
+```hcl
+locals {
+  resources = split("\n---\n", data.template_file.ngnix.rendered)
+}
+
+resource "k8s_manifest" "nginx-deployment" {
+  count = length(local.resources)
+
+  content = local.resources[count.index]
+}
+```
 
 ## Helm workflow
 

--- a/examples/manifests/my-configmap.yaml
+++ b/examples/manifests/my-configmap.yaml
@@ -1,3 +1,5 @@
+---
+
 kind: ConfigMap
 apiVersion: v1
 metadata:

--- a/k8s/resource_k8s_manifest.go
+++ b/k8s/resource_k8s_manifest.go
@@ -3,7 +3,6 @@ package k8s
 import (
 	"context"
 	"fmt"
-	"io"
 	"log"
 	"strings"
 	"time"
@@ -51,9 +50,9 @@ func resourceK8sManifest() *schema.Resource {
 			},
 		},
 		Timeouts: &schema.ResourceTimeout{
-			Create:  schema.DefaultTimeout(3 * time.Minute),
-			Update:  schema.DefaultTimeout(3 * time.Minute),
-			Delete:  schema.DefaultTimeout(3 * time.Minute),
+			Create: schema.DefaultTimeout(3 * time.Minute),
+			Update: schema.DefaultTimeout(3 * time.Minute),
+			Delete: schema.DefaultTimeout(3 * time.Minute),
 		},
 	}
 }
@@ -63,14 +62,9 @@ func resourceK8sManifestCreate(d *schema.ResourceData, config interface{}) error
 	namespace := d.Get("namespace").(string)
 	content := d.Get("content").(string)
 
-	decoder := yaml.NewYAMLOrJSONDecoder(strings.NewReader(content), 4096)
-
-	var object *unstructured.Unstructured
-
-	// TODO: add support for a list of objects?
-	err := decoder.Decode(&object)
-	if err != nil && err != io.EOF {
-		return fmt.Errorf("Failed to unmarshal manifest: %s", err)
+	object, err := contentToObject(content)
+	if err != nil {
+		return err
 	}
 
 	objectNamespace := object.GetNamespace()
@@ -271,14 +265,9 @@ func resourceK8sManifestUpdate(d *schema.ResourceData, config interface{}) error
 
 	content := d.Get("content").(string)
 
-	decoder := yaml.NewYAMLOrJSONDecoder(strings.NewReader(content), 4096)
-
-	var object *unstructured.Unstructured
-
-	// TODO: add support for a list of objects?
-	err = decoder.Decode(&object)
-	if err != nil && err != io.EOF {
-		return fmt.Errorf("Failed to unmarshal manifest: %s", err)
+	object, err := contentToObject(content)
+	if err != nil {
+		return err
 	}
 
 	objectNamespace := object.GetNamespace()
@@ -428,4 +417,21 @@ func resourceK8sManifestImport(d *schema.ResourceData, config interface{}) ([]*s
 	resource.SetId(d.Id())
 
 	return []*schema.ResourceData{&resource}, nil
+}
+
+func contentToObject(content string) (*unstructured.Unstructured, error) {
+	decoder := yaml.NewYAMLOrJSONDecoder(strings.NewReader(content), 4096)
+
+	var object *unstructured.Unstructured
+
+	for {
+		err := decoder.Decode(&object)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to unmarshal manifest: %s", err)
+		}
+
+		if object != nil {
+			return object, nil
+		}
+	}
 }

